#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 942f653ce8e22d562496ce558e5af7ea71ede44c Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Thu, 6 Aug 2020 15:55:57 +0200
Subject: [PATCH] Show about data information from VERSION

This shows the same information as kleopatra
in the about dialog.

GnuPG-Bug-Id: T5011
---
 src/gpgolconfig/gpgoldebugpage.cpp | 44 ++++++++++++++++++++++++++++--
 1 file changed, 42 insertions(+), 2 deletions(-)

diff --git a/src/gpgolconfig/gpgoldebugpage.cpp b/src/gpgolconfig/gpgoldebugpage.cpp
index cfc90e4..cb9900e 100644
--- a/src/gpgolconfig/gpgoldebugpage.cpp
+++ b/src/gpgolconfig/gpgoldebugpage.cpp
@@ -22,6 +22,8 @@
 #include <QLineEdit>
 #include <QStandardPaths>
 #include <QDir>
+#include <QThread>
+#include <Libkleo/GnuPG>

 /* See gpgol/src/debug.h */
 #define DBG_OOM            (1<<1)
@@ -37,6 +39,22 @@ GpgOLDebugPage::GpgOLDebugPage(QWidget *parent):
     load();
 }

+class VersionLoader: public QThread
+{
+    Q_OBJECT
+    void run() override {
+        const auto vers = Kleo::gpg4winVersion();
+        const auto desc = Kleo::gpg4winDescription();
+        const auto longDesc = Kleo::gpg4winLongDescription();
+        emit resultReady(vers, desc, longDesc);
+        deleteLater();
+    }
+signals:
+    void resultReady(const QString &vers,
+                     const QString &desc,
+                     const QString &ldesc);
+};
+
 void GpgOLDebugPage::setupGUI()
 {
     auto baseLay = new QVBoxLayout(this);
@@ -115,12 +133,34 @@ void GpgOLDebugPage::setupGUI()

     baseLay->addWidget(othersGrp);

+    auto aboutGroup = new QGroupBox();
+    auto aboutLay = new QVBoxLayout(aboutGroup);
+    auto descLabel = new QLabel;
+    auto desc2Label = new QLabel;
+    descLabel->setOpenExternalLinks(true);
+    aboutLay->addWidget(descLabel);
+    aboutLay->addWidget(desc2Label);
+    aboutGroup->setVisible(false);
+    baseLay->addWidget(aboutGroup);
+
     auto bugReportLabel = new QLabel(QStringLiteral("<a href=\"https://www.gpg4win.org/reporting-bugs.html\">%1</a>").arg(_("How to report a problem?")));
     bugReportLabel->setOpenExternalLinks(true);
-
     baseLay->addWidget(bugReportLabel);
-
     baseLay->addStretch(1);
+
+    auto loader = new VersionLoader;
+    connect(loader, &VersionLoader::resultReady,
+            this, [descLabel, desc2Label, aboutGroup] (const QString &version, const QString &desc, const QString &desc2) {
+        if (version.isEmpty()) {
+            return;
+        }
+        aboutGroup->setTitle(_("About GpgOL") + QLatin1Char(' ') + version);
+        qDebug() << "Verified version" << version;
+        descLabel->setText(desc);
+        desc2Label->setText(desc2);
+        aboutGroup->setVisible(true);
+    });
+    loader->start();
 }

 static bool loadBool(const char *name, bool defaultVal)
--
2.20.1
